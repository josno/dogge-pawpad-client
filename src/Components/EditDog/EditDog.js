import React, { Component } from "react";

class EditDog extends Component {
	constructor(props) {
		super(props);
		this.state = {
			gender: "",
			age: "",
			tag_number: "",
			microchip: "",
		};
	}
	render() {
		return (
			<div className='basic-dog-details box-flex'>
				<h3 className='info-title'>Basic Details </h3>
				<ul className='dog-info-text details-grid-container'>
					<button onClick={this.props.changeEditMode}>Cancel</button>
					<li className='gender align-details'>Gender: </li>

					<li className='gender-value align-details'>
						<input value={this.props.dogInfo.gender} />
					</li>
					<li className='age align-details'>Age: </li>
					<li className='age-value align-details'>{this.props.dogInfo.age}</li>
					<li className='arrival align-details'>Arrival Date: </li>
					<li className='arrival-value align-details'>
						{this.props.formatDate(this.props.dogInfo.arrival_date)}
					</li>
					<li className='tag align-details'>Tag: </li>
					<li className='tag-value align-details'>
						{this.props.dogInfo.tag_number}
					</li>
					<li className='microchip align-details'>Microchip: </li>
					<li className='microchip-value align-details'>
						{this.props.dogInfo.microchip}
					</li>
					<li className='spayed-neutered align-details'>
						{/* {props.renderSpayedNeutered(props.dogInfo.spayedneutered)} */}
					</li>
				</ul>
			</div>
		);
	}
}

export default EditDog;
