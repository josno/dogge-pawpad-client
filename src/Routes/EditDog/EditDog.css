.updated-by {
	font-size: 0.75em;
	text-align: center;
}

.dog-name-photo {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.shot-instructions {
	font-size: 0.75em;
	font-style: italic;
}

.block {
	display: block;
	font-size: 1em;
	height: 40px;
	padding-left: 10px;
	width: 100%;
}

.bold {
	font-size: 0.8em;
	font-weight: bold;
	color: #f7567c;
}

.fielditem {
	height: 80px;
}
/* 
input[type='radio'],
input[type='checkbox'] {
	margin-left: 20px;
} */

fieldset {
	display: flex;
}

.edit-shot-date {
	margin-left: 10px;
	font-size: 0.8em;
}
