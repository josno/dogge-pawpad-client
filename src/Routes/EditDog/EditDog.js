import React, { Component } from 'react';
import PawPadContext from '../../PawPadContext.js';
import './EditDog.css';
import { Link } from 'react-router-dom';
import DogsApiService from '../../services/api-service';
import EditShots from '../../Components/EditShots/EditShots';
import moment from 'moment';
import Validate from '../../Utils/validation';
import ValidationError from '../../Components/ValidationError/ValidationError';

class EditDog extends Component {
	static contextType = PawPadContext;
	constructor(props) {
		super(props);
		this.state = {
			dogName: '',
			profileImg: '',
			spayedNeutered: false,
			gender: '',
			age: { value: '', touched: false },
			arrivalDate: '',
			updatedBy: '',
			error: ''
		};
		this.formatDate = this.formatDate.bind(this);
		this.handleCheckbox = this.handleCheckbox.bind(this);
		this.handleChange = this.handleChange.bind(this);
		this.updateAge = this.updateAge.bind(this);
		this.handleSubmit = this.handleSubmit.bind(this);
	}

	formatDate = date => {
		const formattedDate = moment(date).format('YYYY-MM-DD');

		return formattedDate;
	};

	handleChange = e => {
		const { name, value } = e.target;
		this.setState({
			[name]: value
		});
	};

	updateAge(age) {
		this.setState({
			age: { value: age, touched: true }
		});
	}

	handleCheckbox = e => {
		const { name } = e.target;

		this.setState(prevState => ({
			[name]: !prevState[name]
		}));
	};

	handleSubmit = e => {
		e.preventDefault();
		const {
			dogName,
			profileImg,
			spayedNeutered,
			gender,
			arrivalDate
		} = this.state;

		const updatedDogObj = {
			dog_name: dogName,
			profile_img: profileImg,
			spayedneutered: spayedNeutered,
			gender: gender,
			age: this.state.age.value,
			arrival_date: new Date(arrivalDate)
		};

		DogsApiService.updateDog(updatedDogObj, this.props.dogId)
			.then(res =>
				this.props.history.push(`/dog-info/${this.props.dogId}`)
			)
			.catch(error => this.setState({ error: error.message }));
	};

	componentDidMount() {
		window.scrollTo(0, 0);
		const { dogId } = this.props;

		DogsApiService.getDogInfo(dogId).then(dogInfo => {
			this.setState({
				dogName: dogInfo.dog_name,
				profileImg: dogInfo.profile_img,
				spayedNeutered: dogInfo.spayedneutered,
				gender: dogInfo.gender,
				age: { value: dogInfo.age, touched: false },
				arrivalDate: dogInfo.arrival_date
			});
		});
	}

	render() {
		return (
			<main className="edit-dog-container">
				<h1 className="form-title">Edit Dog</h1>
				<form className="form-container" onSubmit={this.handleSubmit}>
					<div className="dog-name-photo">
						<h1 htmlFor="name">{this.state.dogName}</h1>
						<img
							id="img-preview"
							src={this.state.profileImg}
							alt="your-pic"
						/>
					</div>
					<h3 className="form-title">Update Information Below </h3>
					<fieldset className="field-item">
						<legend className="bold">Gender:</legend>
						<label htmlFor="male">
							<input
								type="radio"
								name="gender"
								value="Male"
								onChange={this.handleChange}
								id="male"
								checked={
									this.state.gender === 'Male' ? true : false
								}
								required
							/>
							Male
						</label>
						<label htmlFor="female">
							<input
								type="radio"
								name="gender"
								value="Female"
								onChange={this.handleChange}
								id="female"
								checked={
									this.state.gender === 'Female'
										? true
										: false
								}
							/>
							Female
						</label>
					</fieldset>

					<label htmlFor="estimated-age" className="bold">
						Estimated Age
					</label>
					<input
						className="block"
						id="age"
						type="text"
						name="age"
						value={this.state.age.value}
						onChange={e => this.updateAge(e.target.value)}
						required
					/>

					{this.state.age.touched && (
						<ValidationError
							message={Validate.validateAge(this.state.age.value)}
						/>
					)}

					<label htmlFor="arrival" className="bold">
						Estimated Arrival
					</label>
					<input
						className="block"
						id="arrival"
						type="date"
						name="arrivalDate"
						value={this.formatDate(this.state.arrivalDate)}
						onChange={this.handleChange}
					/>

					<EditShots
						key={this.props.dogId}
						dogId={this.props.dogId}
					/>

					<fieldset className="field-item">
						<legend className="bold">Spayed/Neutered</legend>
						<label htmlFor="yes">
							<input
								id="yes"
								type="radio"
								name="spayedNeutered"
								checked={
									this.state.spayedNeutered ? true : false
								}
								onChange={this.handleCheckbox}
							/>
							Yes
						</label>

						<label htmlFor="no">
							<input
								id="no"
								type="radio"
								name="spayedNeutered"
								checked={
									this.state.spayedNeutered ? false : true
								}
								onChange={this.handleCheckbox}
							/>
							No
						</label>
					</fieldset>

					<div className="form-buttons">
						{this.state.error.length > 0 ? (
							<ValidationError
								message={
									'Something wrong happened. Refresh and try again.'
								}
							/>
						) : (
							''
						)}
						<button className="cancel" type="submit">
							<Link className="dog-link" to={'/dogs-list'}>
								Cancel
							</Link>
						</button>

						<button
							className="submit"
							type="submit"
							disabled={Validate.validateAge(
								this.state.age.value
							)}
						>
							Submit
						</button>
					</div>
				</form>
			</main>
		);
	}
}

export default EditDog;
