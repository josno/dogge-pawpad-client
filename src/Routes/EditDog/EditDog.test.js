import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EditDog from './EditDog';

describe('<EditDog />', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<EditDog />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<EditDog />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
