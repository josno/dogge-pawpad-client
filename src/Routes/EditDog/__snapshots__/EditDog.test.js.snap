// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditDog /> renders as expected 1`] = `
<main
  className="edit-dog-container"
>
  <h1
    className="form-title"
  >
    Edit Dog
  </h1>
  <form
    className="form-container"
    onSubmit={[Function]}
  >
    <div
      className="dog-name-photo"
    >
      <h1
        htmlFor="name"
      >
        
      </h1>
      <img
        alt="your-pic"
        id="img-preview"
        src=""
      />
    </div>
    <h3
      className="form-title"
    >
      Update Information Below 
    </h3>
    <fieldset
      className="field-item"
    >
      <legend
        className="bold"
      >
        Gender:
      </legend>
      <label
        htmlFor="male"
      >
        <input
          checked={false}
          id="male"
          name="gender"
          onChange={[Function]}
          required={true}
          type="radio"
          value="Male"
        />
        Male
      </label>
      <label
        htmlFor="female"
      >
        <input
          checked={false}
          id="female"
          name="gender"
          onChange={[Function]}
          type="radio"
          value="Female"
        />
        Female
      </label>
    </fieldset>
    <label
      className="bold"
      htmlFor="estimated-age"
    >
      Estimated Age
    </label>
    <input
      className="block"
      id="age"
      name="age"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="bold"
      htmlFor="arrival"
    >
      Estimated Arrival
    </label>
    <input
      className="block"
      id="arrival"
      name="arrivalDate"
      onChange={[Function]}
      type="date"
      value="Invalid date"
    />
    <fieldset
      className="field-item"
    >
      <legend
        className="bold"
      >
        Required Shots Completed
      </legend>
      <div
        className="update-shot-date-box"
      />
    </fieldset>
    <fieldset>
      <legend
        className="bold"
      >
        Add Optional Shots
      </legend>
      <button
        className="add-shot-btn"
        onClick={[Function]}
        type="button"
      >
        Add A Shot
      </button>
    </fieldset>
    <fieldset
      className="field-item"
    >
      <legend
        className="bold"
      >
        Spayed/Neutered
      </legend>
      <label
        htmlFor="yes"
      >
        <input
          checked={false}
          id="yes"
          name="spayedNeutered"
          onChange={[Function]}
          type="radio"
        />
        Yes
      </label>
      <label
        htmlFor="no"
      >
        <input
          checked={true}
          id="no"
          name="spayedNeutered"
          onChange={[Function]}
          type="radio"
        />
        No
      </label>
    </fieldset>
    <div
      className="form-buttons"
    >
      
      <button
        className="cancel"
        type="submit"
      >
        <a
          className="dog-link"
          href="/dogs-list"
          onClick={[Function]}
        >
          Cancel
        </a>
      </button>
      <button
        className="submit"
        disabled="Age must end in year(s) or month(s)."
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</main>
`;
